src/main/java/com/google/step/snippet/external/StackOverflowClient.java

```diff
@@ -0,0 +1,149 @@
+      "https://api.stackexchange.com/2.2/answers/%s?order"
+          + "=desc&sort=activity&site=stackoverflow&filter=!9_bDE(fI5";
+  // The URL is in the pattern of stackoverlow.com/questions/question_id/title.
+  // The ID_INDEX help retrieve the question_id from parsed URL.
+  private static final int ID_INDEX = 2;
+  private static final String ITEM_PARAMETER = "items";
+  private static final String TITLE_PARAMETER = "title";
+  private static final String BODY_PARAMETER = "body";
+  private static final String CODE_PARAMETER = "code";
+  private static final String ANSWER_ID_PARAMETER = "answer_id";
+  // Set 200 to be the maximum length of description for MVP.
+  private static final int DESCRIPTION_LENGTH_PARAMETER = 200;
+
+  /* This method build the desired card after getting each field. */
+  /* It will return a null card if no valid question is found */
+  public Card search(String url) {
+    String questionId;
+    try {
+      questionId = getQuestionId(url);
+    } catch (URISyntaxException e) {
+      // Return null card if no valid card available.
+      return null;
+    }
+    String answerId = getAnswerId(questionId);
+    if (answerId == null) {
+      return null;
+    }
+    String title = getTitle(questionId);
+    String answerBody = getAnswerBody(answerId);
+    String description = getDescription(answerBody);
+    String code = getCode(answerBody);
+    return new Card(title, code, url, description);
+  }
+
+  /* Get the question id of passed in URL. */
+  private String getQuestionId(String url) throws URISyntaxException {
+    URI uri = new URI(url);
+    // Parse the URL to get the question id.
+    String[] segments = uri.getPath().split("/");
+    String questionId = segments[ID_INDEX];
+    if (!Pattern.matches("[0-9]+", questionId)) {
+      return null;
+    }
+    return questionId;
+  }
+
+  /* Return the question title using question id */
+  private String getTitle(String questionId) {
+    String searchUrl = String.format(SEARCH_URL_TEMPLATE, questionId);
+    return getResponse(searchUrl, TITLE_PARAMETER);
+  }
+
+  /* Return the most voted answer's id. */
+  private String getAnswerId(String questionId) {
+    String questionUrl = String.format(QUESTION_URL_TEMPLATE, questionId);
+    return getResponse(questionUrl, ANSWER_ID_PARAMETER);
+  }
+
+  /* Get the content of the answer and store it in the card. */
+  private String getAnswerBody(String answerId) {
+    String answerUrl = String.format(ANSWER_URL_TEMPLATE, answerId);
+    return getResponse(answerUrl, BODY_PARAMETER);
+  }
+
+  /* Return the description parsed from answer body. */
+  private String getDescription(String body) {
+    Document doc = Jsoup.parse(body);
+    // Combine all description in the answer body.
+    Elements descriptionHtml = doc.select("p");
+    String description = "";
+    for (Element e : descriptionHtml) {
+      description += e.outerHtml();
+    }
+    if (description.length() >= DESCRIPTION_LENGTH_PARAMETER) {
```

Comment:
Consider moving this into the for-loop so you can break early once you've reached the max description length.